### DISCUSSION TITLE
New language

### DISCUSSION THREAD

**Discussion by ghost** [aradzie/keybr.com#224] [Ideas]
I would like to add croatian. How can I do that?

**Comment by aradzie** [comment]
Hi!

I see you added a Croatian translation on POEditor. That's amazing! I have already updated the web site.

To add a training language, the biggest challenge is to find a word frequency dictionary, or a large corpus of text to scan for words.
I have developed my own set of tools for building word frequency dictionaries in [a separate repository](https://github.com/aradzie/keybr.com-corpus).

Over time I developed the following approach:

1. I scan a corpus of text built from [movie subtitles](https://opus.nlpl.eu/OpenSubtitles/en&hr/v2018/OpenSubtitles) for word frequencies. I chose this corpus because movie subtitles are written in a simple contemporary language, without a bias to a specific topic, like politics, news, etc. You can suggest another corpus, if you know a better one, like [the Croatian National Corpus](https://en.wikipedia.org/wiki/Croatian_National_Corpus).
2. I send a list of words to a native speaker to review and remove obscene, profane, triggering words.
3. Then I include the reviewed list of words into keybr to add a new training langauge.

In the coming days I'll build a Croatian word list and send you for review.
